x-pack/filebeat/input/httpjson/config.go

```diff
@@ -125,10 +125,16 @@ func (dc *dateCursorConfig) Validate() error {
 	if dc.DateFormat == "" {
 		return nil
 	}
-	now := time.Now().Format(dc.DateFormat)
-	if _, err := time.Parse(dc.DateFormat, now); err != nil {
+
+	const knownTimestamp = 1602601228 // 2020-10-13T15:00:28+00:00 RFC3339
+
+	// if we can't get back the same date from Format than from Parse
```

Comment:
Is there a requirement for the format to include seconds? Without seconds in the format this test won't pass, right?

Reply:
True, and the same applies for any other element of the date since we can't know the intent of the format. I changed it to just check for parse errors instead, just as a sanity check. Since now it is against a known date we can trigger a parse error from the tests.